src/app/parcels/add/AddParcelForm.tsx

```diff
@@ -0,0 +1,194 @@
+    collectionDate: null,
+    collectionTime: null,
+    collectionCentre: null,
+};
+
+const initialFormErrors: FormErrors = {
+    voucherNumber: Errors.none,
+    packingDate: Errors.initial,
+    timeOfDay: Errors.initial,
+    shippingMethod: Errors.initial,
+    collectionDate: Errors.initial,
+    collectionTime: Errors.initial,
+    collectionCentre: Errors.initial,
+};
+
+const mergeDateAndTime = (date: Date, time: Date): Date => {
+    return new Date(
+        date.getFullYear(),
+        date.getMonth(),
+        date.getDate(),
+        time.getHours(),
+        time.getMinutes()
+    );
+};
+
+const AddParcelForm: React.FC<{ id: string }> = (props: { id: string }) => {
+    const router = useRouter();
+    const [fields, setFields] = useState(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const formSections =
+        fields.shippingMethod === "Collection"
+            ? withCollectionFormSections
+            : noCollectionFormSections;
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitErrorMessage("");
+        setSubmitDisabled(true);
+        let inputError;
+        if (fields.shippingMethod === "Collection") {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors);
+        } else {
+            inputError = checkErrorOnSubmit(formErrors, setFormErrors, [
+                "voucherNumber",
+                "packingDate",
+                "timeOfDay",
+                "shippingMethod",
+            ]);
+        }
+        if (inputError) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+        if (fields.packingDate === null || fields.timeOfDay === null) {
+            setSubmitError(Errors.external);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const packingDateTime = mergeDateAndTime(
+            new Date(fields.packingDate),
+            new Date(fields.timeOfDay)
+        );
+
+        let collectionDateTime = null;
+        if (fields.shippingMethod === "Collection") {
+            if (fields.collectionDate === null || fields.collectionTime === null) {
+                setSubmitDisabled(false);
+                setSubmitError(Errors.external);
+                return;
+            }
+            collectionDateTime = mergeDateAndTime(
+                new Date(fields.collectionDate),
+                new Date(fields.collectionTime)
+            ).toISOString();
+        }
+
+        const isDelivery = fields.shippingMethod === "Delivery";
+
+        const formToAdd = {
+            client_id: props.id,
```

Comment:
you can use just `id` because of the `{id}` in the input to the function

Reply:
Changed